pkg/volume/fc/fc.go

```diff
@@ -249,6 +320,77 @@ func (c *fcDiskUnmounter) TearDownAt(dir string) error {
 	return diskTearDown(c.manager, *c, dir, c.mounter)
 }
 
+// Block Volumes Support
+type fcDiskMapper struct {
+	*fcDisk
+	readOnly   bool
+	volumeMode v1.PersistentVolumeMode
+	volumePath string
+	mounter    mount.Interface
+}
+
+var _ volume.BlockVolumeMapper = &fcDiskMapper{}
+
+func (b *fcDiskMapper) CanBlockMap() error {
+	return nil
+}
+
+func (b *fcDiskMapper) SetUpDevice() error {
+	return b.SetUpDeviceAt(b.GetMapPath())
+}
+
+func (b *fcDiskMapper) SetUpDeviceAt(dir string) error {
+	// diskSetUp checks mountpoints and prevent repeated calls
+	err := deviceSetUp(b.manager, *b, dir)
+	if err != nil {
+		glog.Errorf("fc: failed to setup")
+	}
+	return err
+}
+
+func (b *fcDiskMapper) GetVolumePath() string {
+	devicePath := ""
+	mounter := fcDiskMounter{
+		fcDisk: &fcDisk{
+			wwns:  b.wwns,
+			lun:   b.lun,
+			wwids: b.wwids,
+			io:    b.io,
+		},
+	}
+	disk, dm := searchDisk(mounter)
+	if dm != "" {
+		devicePath = dm
+	} else {
+		devicePath = disk
+	}
+	return devicePath
+}
+
+func (b *fcDiskMapper) GetVolumeMode() v1.PersistentVolumeMode {
```

Comment:
This method does not seem to be called.

Reply:
@jsafrane This method is currently called from makeBlockVolumes() in kubelet_pod.go but is under discussion on the other thread. May be we can remove.

https://github.com/kubernetes/kubernetes/pull/51494

Reply:
Do you have a branch or complete PR with all the code that would compile and run? Cross-reviewing several partial PRs is hard.

Reply:
@jsafrane 
We are developing this feature in parallel, therefore we don't have complete PR.

But let me try something to review easily.
I understand these separated PRs are too hard to review without compile and run them.

Reply:
You can cherry-pick the other PRs into your branch temporarily, and then remove them when they get merged.

Reply:
@msau42 @jsafrane 
yeah, good idea. will do.